main.go

```diff
@@ -630,15 +633,26 @@ func _main(ctx context.Context) error {
 		return nil
 	}
 
+	log.Info("Mainchain sync complete.")
+
+	// Ensure all side chains known by dcrd are also present in the base DB
+	// and import them if they are not already there.
+	if cfg.ImportSideChains {
+		log.Info("Primary DB -> Now retrieving side chain blocks from dcrd...")
+		err := baseDB.ImportSideChains(collector)
```

Comment:
It's a minor issue in most cases, but the approach of doing sqlite first then possibly postgres is inefficient. That is,

```go
if cfg.ImportSideChains {
	// retrieve all side chains and import into sqlite
}
if usePG && cfg.ImportSideChains {
	// retrieve all side chains and import into sqlite
}
```

vs.

```go
if cfg.ImportSideChains {
	// retrieve all side chains

	// import into sqlite

	if usePG {
		// import into postgres
	}
}
```

Changing to the latter requires some refactoring, so it's not a big deal right now.  Just take note.

Reply:
This was part of the reason I tried to implement this as a single function call in main.go. Is there any reason both of these sidechain import routines can't be implemented in asynchronously?

Reply:
It's not an issue to do it separately for sqlite and pg, but the retrieval gets run twice.  Usually this is just a handful of blocks, so no worries.  On testnet3 there was a crazy 4k block reorg early on, and I my node still remembers it -- which is great for testing these things.  But this is certainly not something that will be an issue on mainnet.

Reply:
K. Maybe as a separate issue, but I'm happy to do this. Probably have a wrapper function that pulls the blocks once and feeds them through a checker method of each DB object. Can that be run as a goroutine?

Reply:
I don't see why they can't be run concurrently.